Sources/Action.swift

```diff
@@ -213,43 +223,47 @@ extension Action: BindingTargetProvider {
 }
 
 extension Action where Input == Void {
-	/// Initializes an action that uses an `Optional` property for its input,
-	/// and is disabled whenever the input is `nil`. When executed, a `SignalProducer`
-	/// is created with the current value of the input.
+	/// Initializes an `Action` that uses a property of optional as its state.
+	///
+	/// When the `Action` is asked to start the execution, a unit of work — represented by
+	/// a `SignalProducer` — would be created by invoking `workProducer` with the latest
+	/// value of the state.
+	///
+	/// If the property holds a `nil`, the `Action` would be disabled until it is not
+	/// `nil`.
 	///
 	/// - parameters:
-	///   - input: An `Optional` property whose current value is used as input
-	///            whenever the action is executed. The action is disabled
-	///            whenever the value is `nil`.
-	///   - execute: A closure to return a new `SignalProducer` based on the
-	///              current value of `input`.
-	public convenience init<P: PropertyProtocol, T>(input: P, _ execute: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T? {
-		self.init(state: input, enabledIf: { $0 != nil }) { input, _ in
-			execute(input!)
+	///   - state: A property of optional to be the state of the `Action`.
+	///   - workProducer: A closure that produces a unit of work, as `SignalProducer`, to
+	///                   be executed by the `Action`.
+	public convenience init<P: PropertyProtocol, T>(state: P, _ workProducer: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T? {
+		self.init(state: state, enabledIf: { $0 != nil }) { state, _ in
+			workProducer(state!)
 		}
 	}
 
-	/// Initializes an action that uses a property for its input. When executed,
-	/// a `SignalProducer` is created with the current value of the input.
+	/// Initializes an `Action` that uses a property as its state.
+	///
+	/// When the `Action` is asked to start the execution, a unit of work — represented by
+	/// a `SignalProducer` — would be created by invoking `workProducer` with the latest
+	/// value of the state.
 	///
 	/// - parameters:
-	///   - input: A property whose current value is used as input
-	///            whenever the action is executed.
-	///   - execute: A closure to return a new `SignalProducer` based on the
-	///              current value of `input`.
-	public convenience init<P: PropertyProtocol, T>(input: P, _ execute: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T {
-		self.init(input: input.map(Optional.some), execute)
+	///   - state: A property to be the state of the `Action`.
+	///   - workProducer: A closure that produces a unit of work, as `SignalProducer`, to
+	///                   be executed by the `Action`.
+	public convenience init<P: PropertyProtocol, T>(state: P, _ workProducer: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T {
```

Comment:
> I wouldn't say that this property describes the state of the `Action`. It's the inputs that are used—consumed in a reactive way.

The point is though the term _input_ is already bound to the caller-facing `apply()`, which does not take any input in this case. So calling this `input` leads to two interpretations of `input`, which do not seem nice for documentation purpose.

Moreover, it is no difference from `init(state:enabledIf:_:)` after all, just that it has a default `enabledIf` closure that asserts when the value is `nil`.

> `workProducer` is a bit misleading because it's not a producer itself—it's a closure that returns a producer. So I think `execute` is a preferable name here as well.

I can't defend `workProducer` in this regard. But `execute` doesn't seem right, since it doesn't really execute but returns something to be executed. `workFactory`, `producerFactory` or `workCustomizer` are possible alternatives.